python/pyarrow/_compute.pyx

```diff
@@ -1168,6 +1168,42 @@ class SliceOptions(_SliceOptions):
         self._set_options(start, stop, step)
 
 
+cdef class _ListSliceOptions(FunctionOptions):
+    cpdef _set_options(self, start, stop=None, step=1, return_fixed_size_list=True):
+        cdef:
+            CListSliceOptions* opts
+        if stop is None:
+            opts = new CListSliceOptions(start,
+                                         <optional[int64_t]>nullopt,
+                                         step, return_fixed_size_list)
+        else:
+            opts = new CListSliceOptions(start,
+                                         <optional[int64_t]>(<int64_t>stop),
+                                         step, return_fixed_size_list)
+        self.wrapped.reset(opts)
+
+
+class ListSliceOptions(_ListSliceOptions):
+    """
+    Options for list array slicing.
+
+    Parameters
+    ----------
+    start : int
+        Index to start slicing inner list elements (inclusive).
+    stop : Optional[int], default None
+        If given, index to stop slicing at (exclusive).
+        If not given, slicing will stop at the end. (NotImplemented)
+    step : int, default 1
+        Slice step.
+    return_fixed_size_list : bool, default True
+        Whether to return a FixedSizeListArray.
```

Comment:
We should also include here the " If stop is after a list element's length, nulls will be appended to the requested slice size." as you have in the C++ docstring